http-client/src/main/java/io/airlift/http/client/jetty/BufferingResponseListener.java

```diff
@@ -126,10 +121,10 @@ public String getEncoding()
      */
     public byte[] getContent()
     {
-        if (buffer == null) {
+        if (buffer.getLength() == 0) {
             return new byte[0];
         }
-        return BufferUtil.toArray(buffer);
+        return buffer.takeByteBuffer().array();
```

Comment:
This may return more data than what's represented by the ByteBuffer bounds, no? Is that what we want here?

Reply:
takeByteBuffer does the copy to a new sized byte buffer